### ISSUE TITLE
Add autobatch as default in local YOLOv5 runs

### ISSUE THREAD

**Issue by Powercube7** [ultralytics/yolov5#7540]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

From what I can see, the Ultralytics Hub uses the training of the default repo, meaning the autoBatch functionality is already implemented. However, this is not a default in local clones of the repo. I've had constant difficulty with trial and error using the batch section, so this becoming a default would dramatically decrease the time I spend trying to fix the errors because of batches being too large

### Use case

This feature could be used universally, saving people a lot of time when trying to find the right batch for the job

- The feature could be a default or be enabled if the flag --batch "Auto" is passed when starting training
- The feature could be manually overridden by using the --batch flag with the desired quantity per batch

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Powercube7 that's not a bad idea and I think we want to trend in that direction, but AutoBatch does have a few limitations, mainly that it does not work with Multi-GPU currently. Maybe we could make the AutoBatch calculation on RANK 0 with a barrier until it's complete, then assign the autobatch size to an environment variable that the other ranks could read. This is an example of a barrier action where all ranks wait until rank zero finishes:
https://github.com/ultralytics/yolov5/blob/4cac1cfc8f66569b066623584d9cf3fe06cf54f4/train.py#L106-L109

If you'd like to try to take the lead on Multi-GPU AutoBatch, that's really the last puzzle piece we need in place before we can default it to auto, which I agree would be nice to have.

Please see our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/) to get started.
